serviceWorker.js

```diff
@@ -0,0 +1,108 @@
+// Set to false to disable logging.
+const verbose = true;
+
+async function handleFetch(request) {
+	// If the request is cache-only and the mode is not same-origin, return nothing
+	if (request.cache === "only-if-cached" && request.mode !== "same-origin") {
+		return;
+	}
+
+	// Perform the original fetch request and store the result
+	const r = await fetch(request).catch(e => verbose && console.error(e));
+
+	// If the fetch request returns status 0, return the response
+	if (r.status === 0) {
+		return r;
+	}
+
+	// We add headers to the original response its headers, in order to enable cross-origin-isolation. And make it independent of the server config.
+	const newHeaders = new Headers(r.headers);
+
+	// COEP & COOP for cross-origin-isolation
+	newHeaders.set("Cross-Origin-Embedder-Policy", "require-corp");
+	newHeaders.set("Cross-Origin-Opener-Policy", "same-origin");
+
+	newHeaders.set("Cross-Origin-Resource-Policy", "cross-origin");
+
+	/** 
+	 * This is a hacky fix, due to cheerpOS relying on the response.url being set to the resolved url (after a redirect)
+	 * when we construct a new response, it will not have the correct url set, and the serviceworker respondwith method will set the url
+	 * to the value of event.request.url; see https://developer.mozilla.org/en-US/docs/Web/API/FetchEvent/respondWith.
+	 * In case of a redirect, we therefore set the location url to the resolved response url, and set statuscode to 301 in our new 
+	 * Response object, this then bounces the request back to the serviceworker from Cheerpos, with the request.url being set to the resolved url, therefore responseWith will set it properly in our new response.
+	 */
+	if (r.redirected === true)
+		newHeaders.set("location", r.url);
+
+	// In case of a redirection, we set the status to 301, and body to null, in order to not transfer too much data needlessly
+	const moddedResponse = new Response(r.redirected === true ? null : r.body, {
+		headers: newHeaders,
+		status: r.redirected === true ? 301 : r.status,
+		statusText: r.statusText,
+	});
+	return moddedResponse;
+}
+
+function serviceWorkerInit() {
+	// Init the service worker.
+	self.addEventListener("install", () => self.skipWaiting());
+	self.addEventListener("activate", e => e.waitUntil(self.clients.claim()));
+
+	// Listen for fetch requests and call handleFetch function.
+	self.addEventListener("fetch", function (e) {
+		try {
+			e.respondWith(handleFetch(e.request));
+		} catch (err) {
+			verbose && console.log("Serviceworker respondWith failed with ERROR:" + err);
+		}
+	});
+}
+
+function doRegister() {
+	navigator.serviceWorker.register(window.document.currentScript.src)
+		.then((registration) => {
+			verbose && console.log("Service Worker registered", registration.scope);
+
+			// Reload the page to make use of an updated service worker
+			registration.addEventListener("updatefound", () => {
```

Comment:
It seems to me that forcing that update would be disruptive on running sessions. Also, we don't expect to update this script ever at all